File: math/number-theory/digit_sum.go
// Copyright https://github.com/microwind
// Digit Sum

package main

import "fmt"

func digitSum(n int) int {
	sum := 0
	for n > 0 {
		sum += n % 10
		n /= 10
	}
	return sum
}

func main() {
	n := 12345
	fmt.Printf("Digit sum of %d = %d\n", n, digitSum(n))
}

File: math/number-theory/DigitSum.java
/**
 * Copyright https://github.com/microwind
 * Digit Sum
 */

class DigitSum {
    public static int digitSum(int n) {
        int sum = 0;
        while (n > 0) {
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }
    
    public static void main(String[] args) {
        int n = 12345;
        System.out.println("Digit sum of " + n + " = " + digitSum(n));
    }
}
